### DISCUSSION TITLE
Scaling (minimizing content) on PDF

### DISCUSSION THREAD

**Discussion by Begimkins** [pymupdf/PyMuPDF#3986] [Looking for help] [answered]
Hello! Wondering if it is possible to minimize or scale down the content of my pdf when exported. I am trying to add header and footer text but that's overlapping with the original content

**Comment by JorjMcKie** [comment]
What do you mean by "exporting"?

**Reply by Begimkins** [reply]
when you download the pdf - export here is unnecessary. is it possible, overall, to minimize the original content by like 5% from top and bottom to make sure header and footer doesn't overwrite the original content?

**Reply by JorjMcKie** [reply]
Still don't understand:
How can a downloaded PDF overwrite original content? Which original?

**Reply by Begimkins** [reply]
here is my pdf and everything you see on it i refer as "original content" that i retrieve from aws s3 buckets.
i case I add a header, it is inserted on top-right corner 
<img width="1087" alt="Screenshot 2024-10-23 at 1 57 50 PM" src="https://github.com/user-attachments/assets/f5cef0d7-06b9-4eac-bcca-5d24b804b449">

here. I added header manually to show what's happening. 
<img width="471" alt="Screenshot 2024-10-23 at 2 00 05 PM" src="https://github.com/user-attachments/assets/f2bedbb9-2361-4b22-b479-15febbe13218">

I want to scale my original content by -5% so that header or footer doesn't overlap with the content of my pdf

**Reply by JorjMcKie** [reply, marked as answer]
Aha! All this has nothing to do with downloading or not. I was confused by this distracting buzzword.

Here is what you need to do:
1. What you are actually doing is making a copy of pages by inserting them into pages of a new PDF where they take 5% less room than before.
2. When reading an original page, determine its width / height.
3. Make a same-size page in the output PDF.
4. Define a smaller rectangle on output page and insert source page into this rectangle. The scaling will take place atuoamtically in this step.
5. Insert header or footer as desired in target page.

```python
src = pymupdf.open("input.pdf")
doc = pymupdf.open()  # output PDF
for spage in src:  # read source pages
    w, h = spage.rect.br  # widht / height of source page
    page = doc.new_page(width=w, height=h)  # target page
    nw = w * 0.95  # 5% smaller target width
    nh = h * 0.95
    top = (h - nh) / 2  # top coordinate
    left = (w - nw) / 2  # left coord
    tar_rect = pymudf.Rect(left, top, left + nw, top + nh)
    page.show_pdf_page(tar_rect, src, spaage.number)  # insert source page
```

**Reply by Begimkins** [reply]
Amazing, sorry for the buzzword - i will be more accurate next time ❤️ Thank you!

**Reply by JorjMcKie** [reply]
From looking at your example input:
It might be a Form PDF (i.e. containing fillable form fields).
With my method above, the result would no longer be like that. Look good, but fields are no longer available for filling them out.

**Reply by Begimkins** [reply]
Yeah that's alright, forms come filled in and will no longer need to be re-filled, so that worked great! I have just finished exporting one of my pdfs and the result is exactly what I needed
